src/Workspaces/Core/Portable/Workspace/Solution/ProjectChanges.cs

```diff
@@ -130,6 +130,28 @@ public IEnumerable<DocumentId> GetChangedDocuments()
             }
         }
 
+        /// <summary>
+        /// The difference between this GetTextChangedDocuments() and GetChangedDocuments() is:
+        /// GetTextChangedDocuments only returns documents that have text changes
+        /// GetChangedDocuments returns documents that have any changes, i.e. ParseOption, filename, SourceCodeKind ...
+        /// https://github.com/dotnet/roslyn/issues/18199
+        /// </summary>
+        /// <returns></returns>
+        public IEnumerable<DocumentId> GetTextChangedDocuments()
```

Comment:
is this better than having flag in GetChangedDocument? also, we should have same change for AdditionalDocuments as well.

Reply:
hmmm.. what happen if additional files are in source control. and people changed additional files? will it do EnsureEditableDocuments?

Reply:
updated

Reply:
will deal with the additional files scenario in https://github.com/dotnet/roslyn/issues/26254

Reply:
@heejaechang, looks like I have to remove GetChangedDocuments() from PublicAPI.Shipped.txt and add it to the unshipped.txt, is that correct?